test/sample-test/run_test.sh

```diff
@@ -101,242 +103,217 @@ while [ "$1" != "" ]; do
-chmod +x ~/bin/argo
-
-echo "Run the sample tests..."
-
-# Run the tests
-if [ "$TEST_NAME" == 'tf-training' ]; then
-  SAMPLE_KUBEFLOW_TEST_RESULT=junit_SampleKubeflowOutput.xml
-  SAMPLE_KUBEFLOW_TEST_OUTPUT=${RESULTS_GCS_DIR}
-
-  #TODO: convert the sed commands to sed -e 's|gcr.io/ml-pipeline/|gcr.io/ml-pipeline-test/' and tag replacement. 
-  # Compile samples
-  cd ${BASE_DIR}/samples/core/kubeflow-tf
-
-  if [ -n "${DATAFLOW_TFT_IMAGE}" ];then
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-dataflow-tft:\([a-zA-Z0-9_.-]\)\+|${DATAFLOW_TFT_IMAGE}|g" kubeflow-training-classification.py
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-kubeflow-tf-trainer:\([a-zA-Z0-9_.-]\)\+|${KUBEFLOW_DNNTRAINER_IMAGE}|g" kubeflow-training-classification.py
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-dataflow-tf-predict:\([a-zA-Z0-9_.-]\)\+|${DATAFLOW_PREDICT_IMAGE}|g" kubeflow-training-classification.py
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-local-confusion-matrix:\([a-zA-Z0-9_.-]\)\+|${LOCAL_CONFUSIONMATRIX_IMAGE}|g" kubeflow-training-classification.py
-  fi
-
-  dsl-compile --py kubeflow-training-classification.py --output kubeflow-training-classification.zip
+################################################################################
+# Utility function to setup working dir, input and output locations.
+# Globals:
+#   SAMPLE_TEST_RESULT, test output file.
+#   SAMPLE_TEST_OUTPUT, test output location.
+# Arguments:
+#   Test name.
+################################################################################
+preparation() {
+  SAMPLE_TEST_RESULT="junit_Sample$1Output.xml"
+  SAMPLE_TEST_OUTPUT=${RESULTS_GCS_DIR}
+  WORK_DIR=${BASE_DIR}/samples/core/$1
+  cd ${WORK_DIR}
+}
 
+################################################################################
+# Utility function to generate formatted result, and move it to a place for PROW
+# to pick up.
+# Arguments:
+#   Test name.
+################################################################################
+check_result() {
   cd "${TEST_DIR}"
-  python3 run_kubeflow_test.py --input ${BASE_DIR}/samples/core/kubeflow-tf/kubeflow-training-classification.zip --result $SAMPLE_KUBEFLOW_TEST_RESULT --output $SAMPLE_KUBEFLOW_TEST_OUTPUT --namespace ${NAMESPACE}
+  python3 run_sample_test.py --input ${WORK_DIR}/$1.yaml \
+  --result ${SAMPLE_TEST_RESULT} --output ${SAMPLE_TEST_OUTPUT} --testname $1 \
+  --namespace ${NAMESPACE}
 
   echo "Copy the test results to GCS ${RESULTS_GCS_DIR}/"
-  gsutil cp ${SAMPLE_KUBEFLOW_TEST_RESULT} ${RESULTS_GCS_DIR}/${SAMPLE_KUBEFLOW_TEST_RESULT}
-elif [ "$TEST_NAME" == "tfx" ]; then
-  SAMPLE_TFX_TEST_RESULT=junit_SampleTFXOutput.xml
-  SAMPLE_TFX_TEST_OUTPUT=${RESULTS_GCS_DIR}
-  
-  # Compile samples
-  cd ${BASE_DIR}/samples/core/tfx
-
-  dsl-compile --py taxi-cab-classification-pipeline.py --output taxi-cab-classification-pipeline.yaml
+  gsutil cp ${SAMPLE_TEST_RESULT} ${RESULTS_GCS_DIR}/${SAMPLE_TEST_OUTPUT}
```

Comment:
Should this be
gsutil cp ${SAMPLE_TEST_RESULT} ${RESULTS_GCS_DIR}/${SAMPLE_TEST_RESULT}
?

Reply:
Good catch. Thanks!